src/shared/components/search.tsx

```diff
@@ -466,6 +466,10 @@ export class Search extends Component<any, SearchState> {
         <HtmlTags
           title={this.documentTitle}
           path={this.context.router.route.match.url}
+          canonicalPath={
+            this.context.router.route.match.url +
+            this.context.router.route.location.search
```

Comment:
I thought `url` already contains the query params?

Reply:
I would have thought so from the name, but apparently not:

```js
console.log(this.context.router.route);
```

```json
{
  "location": {
    "pathname": "/search",
    "search": "?q=test&type=All&listingType=All&page=1&sort=TopAll",
    "hash": "",
    "state": null,
    "key": "1hxl4e7m"
  },
  "match": {
    "isExact": true,
    "params": {},
    "path": "/search",
    "url": "/search"
  }
}
```

A separate change might want to change the OpenGraph URLs some places as it indeed always links without the query parameters.